src/utils/vm_mount.cpp

```diff
@@ -105,12 +107,27 @@ mp::VMMount::VMMount(const std::string& sourcePath,
     if (errors.size())
         throw std::runtime_error(
             fmt::format("Mount cannot apply mapping with duplicate ids:{}", fmt::to_string(errors)));
+
+    resolve_source_path();
 }
 
 mp::VMMount::VMMount(const QJsonObject& json) : VMMount{parse_json(json)} // delegate on copy ctor
 {
 }
 
+void mp::VMMount::resolve_source_path()
```

Comment:
I think that makes more sense, something truly exceptional has to occur for `weakly_canonical` to fail. My only reason for using the error code version was because all the other std functions in file_ops use that overload.